internal/host/validator.go

```diff
@@ -1714,3 +1714,58 @@ func (v *validator) serviceCanConnectToSpokeKubeAPI(c *validationContext) (Valid
 
 	return ValidationSuccess, "Host was able to connect to KubeAPI of the spoke cluster for Day2 operations"
 }
+
+func (v *validator) noIPCollisions(c *validationContext) (ValidationStatus, string) {
+
+	// Obtain the connectivity report.
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending, fmt.Sprintf("Awaiting connectivity report for host %s", c.host.ID)
+	}
+	connectivityReport, err := hostutil.UnmarshalConnectivityReport(c.host.Connectivity)
+	if err != nil {
+		message := fmt.Sprintf("Unable to unmarshall host connectivity for %s: %s", c.host.ID, err)
+		v.log.Errorf(message)
+		return ValidationError, message
+	}
+
+	collidingIPSWithMacs := getCollidingIPs(connectivityReport)
```

Comment:
My understanding is that this would check if node IPs are required to be unique, right? Do we want to make sure that API and Ingress VIPs are not yet assigned too?

Reply:
Right, probably lets leave it like this. I've seen some installs using node address as API VIP, which will fail the install and was wondering if this validation can protect us from such cases.

Reply:
I think there should be another validation should take care of this, this is certainly something worth looking into as the focus of another ticket.